api/v1alpha1/drpolicy_types.go

```diff
@@ -7,31 +7,41 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// nolint:lll
+
 // DRPolicySpec defines the desired state of DRPolicy
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.schedulingInterval) || has(self.schedulingInterval)", message="schedulingInterval is required once set"
```

Comment:
I think the pattern we need to use is [Immutability upon object creation](https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutability-upon-object-creation)

For example with the current scheme a user would be allowed to set an empty value to begin with and then be allowed to modify it to a valid value, the immutability starts here and prevents further modification.

What we want is immutable post creation, so if the policy started with an empty value it should remain so.

Reply:
updated the code with [Immutability upon object creation](https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutability-upon-object-creation) for `schedulingInterval` and `drclusters` only.